pkg/minikube/bootstrapper/bootstrapper.go

```diff
@@ -58,3 +67,63 @@ func GetCachedImageList(version string, bootstrapper string) []string {
 		return []string{}
 	}
 }
+
+// CacheBinaries downloads binaries that will be used by UpdateCluster
+func CacheBinaries(cr CommandRunner, k8s config.KubernetesConfig) error {
+	var g errgroup.Group
+	for _, bin := range []string{"kubelet", "kubeadm"} {
+		bin := bin
+		g.Go(func() error {
+			path, err := maybeDownloadAndCache(bin, k8s.KubernetesVersion)
+			if err != nil {
+				return errors.Wrapf(err, "downloading %s", bin)
+			}
+			if cr == nil {
+				return nil
+			}
+			f, err := assets.NewFileAsset(path, "/usr/bin", bin, "0641")
+			if err != nil {
+				return errors.Wrap(err, "new file asset")
+			}
+			if err := cr.Copy(f); err != nil {
+				return errors.Wrapf(err, "copy")
+			}
+			return nil
+		})
+	}
+	return g.Wait()
+}
+
+func maybeDownloadAndCache(binary, version string) (string, error) {
+	targetDir := constants.MakeMiniPath("cache", version)
+	targetFilepath := path.Join(targetDir, binary)
+
+	url := constants.GetKubernetesReleaseURL(binary, version)
+
+	_, err := os.Stat(targetFilepath)
+	// If it exists, do no verification and continue
+	if err == nil {
+		glog.Infof("Not caching binary, using %s", url)
+		return targetFilepath, nil
+	}
+	if !os.IsNotExist(err) {
+		return "", errors.Wrapf(err, "stat %s version %s at %s", binary, version, targetDir)
+	}
+
+	if err = os.MkdirAll(targetDir, 0777); err != nil {
```

Comment:
Would 0700 also work?

Reply:
Maybe, but I didn't actually change that line.